### PR TITLE
Adding compound aggregate for uddsketch

### PR DESCRIPTION
**Pull request by WireBaron**
This change also fixes some errors in the udd sketch combining code.

### CODE REVIEW

File: crates/udd-sketch/src/lib.rs [hunk 1]
```diff
@@ -47,8 +47,8 @@ impl SketchHashKey {
         use SketchHashKey::*;
 
         match *self {
-            Negative(x) => Negative(if x < 0 {x-1} else {x} /2),
-            Positive(x) => Positive(if x < 0 {x-1} else {x} /2),
+            Negative(x) => Negative(if x > 0 {x+1} else {x} /2),
+            Positive(x) => Positive(if x > 0 {x+1} else {x} /2),
```
Comment (JLockerman) [inline crates/udd-sketch/src/lib.rs:51]: could you explain the change?

File: crates/udd-sketch/src/lib.rs [hunk 2]
```diff
@@ -292,7 +292,7 @@ impl UDDSketch {
             self.compact_buckets();
         }
 
-        for entry in other.buckets.iter() {
+        for entry in target.buckets.iter() {
```
Comment (JLockerman) [inline crates/udd-sketch/src/lib.rs:295]: maybe we should just do `let other = other.clone()` above, so the original `other` is no longer reachable, should prevent bugs like this in the future 🤔

**Review by JLockerman — APPROVED** [review]

File: crates/udd-sketch/src/lib.rs (hunk 1 above)
Comment (WireBaron) [reply crates/udd-sketch/src/lib.rs:50]: Certainly, the problem here is that I was combining bins the wrong direction...so I was pairing bins like this: ..., (-2, -1), (0, 1), (2, 3), ... when it should have been ..., (-3, -2), (-1, 0), (1, 2), ...

I've updated the comment on this function indicating the order of combination.

File: crates/udd-sketch/src/lib.rs (hunk 2 above)
Comment (WireBaron) [reply crates/udd-sketch/src/lib.rs]: Good point, made this change

File: extension/src/uddsketch.rs [hunk 3]
```diff
@@ -335,4 +370,57 @@ mod tests {
...
+                
+            let (test_value, test_error) = client
+                .select("SELECT \
+                    timescale_analytics_experimental.quantile(uddsketch, 0.9), \
+                    timescale_analytics_experimental.error(uddsketch) \
+                    FROM composite", None, None)
+                .first()
+                .get_two::<f64, f64>();
+
+            apx_eql(test_value.unwrap(), value.unwrap(), 0.0001);
```
Comment (davidkohn88) [inline extension/src/uddsketch.rs:421]: I think for uddsketch, the merges shouldn't matter and the result is deterministic whether or not we have the intermediate result, so the apx_eql shouldn't be necessary?

**Pull request merged**